x/interchainstaking/keeper/keeper.go

```diff
@@ -259,6 +259,14 @@ func (k *Keeper) SetValidatorForZone(ctx sdk.Context, zone *types.Zone, data []b
 		return err
 	}
 
+	// In case validator is jailed
+	if validator.IsJailed() {
```

Comment:
This should also check that the validator wasn’t previously jailed, so we don’t keep resending the same request

Reply:
I collected all the logic of the process jailed in [ValidatorSigningInfosCallback](https://github.com/quicksilver-zone/quicksilver/blob/d980e666d07ec0cc60ddd935b1f40566e941665a/x/interchainstaking/keeper/callbacks.go#L258) so I think we will not resend the same request

Reply:
Because if the validator wasn’t previously jailed, we also check if it now was jailed or was tombstones to update info right?

Reply:
The logic now will make the infinite loop when `validator` was jailed, Im checking now